### PR TITLE
Napkins

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Added napkins and drum to hold them.

## Why / Balance
My face dirty.

## How to test
Spaewn in a service techfab, wood and steel.
Print a drum and some napkins.
Insert napkins into drum
Get napkins out
Eat napkins (I mean use).

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame,

image

## Breaking changes
N/A

**Changelog**
:cl: dvir01
- add: Added napkins and napkin drum.

### CODE REVIEW

**Review by ErhardSteinhauer — APPROVED** [review]
do be working

File: Content.Server/Nutrition/Components/FoodComponent.cs [hunk 1]
```diff
@@ -74,4 +74,11 @@ public sealed partial class FoodComponent : Component
     /// </summary>
     [DataField, ViewVariables(VVAccess.ReadWrite)]
     public bool RequireDead = true;
+
+    /// <summary>
+    /// Frontier - Show eating popup.
+    /// </summary>
+    [DataField]
+    public bool ShowEatPopup;
```
Comment (whatston3) [inline Content.Server/Nutrition/Components/FoodComponent.cs:82]: Default to true, maintain existing behaviour.

File: Resources/Textures/_NF/Objects/Specific/Service/napkindrum.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    "x": 32,
+    "y": 32
+  },
+  "states": [
+    {
+      "name": "drum0"
+    },
+    {
+      "name": "drum1"
+    }
```
Comment (whatston3) [inline Resources/Textures/_NF/Objects/Specific/Service/napkindrum.rsi/meta.json:15]: No 1-napkin state. :(

**Review by whatston3 — COMMENTED** [review]
Looks alright, will test, might make a few small edits.

File: Content.Server/Nutrition/EntitySystems/FoodSystem.cs [hunk 3]
```diff
@@ -267,16 +270,21 @@ private void OnDoAfter(Entity<FoodComponent> entity, ref ConsumeDoAfterEvent arg
...
             var userName = Identity.Entity(args.User, EntityManager);
-            _popup.PopupEntity(Loc.GetString("food-system-force-feed-success", ("user", userName), ("flavors", flavors)), entity.Owner, entity.Owner);
-
-            _popup.PopupEntity(Loc.GetString("food-system-force-feed-success-user", ("target", targetName)), args.User, args.User);
+            if (entity.Comp.ShowEatPopup) // Frontier
+            {
+                _popup.PopupEntity(Loc.GetString("food-system-force-feed-success", ("user", userName), ("flavors", flavors)), entity.Owner, entity.Owner);
 
+                _popup.PopupEntity(Loc.GetString("food-system-force-feed-success-user", ("target", targetName)), args.User, args.User);
+            }
```
Comment (whatston3) [inline Content.Server/Nutrition/EntitySystems/FoodSystem.cs:278]: Here's a thought.
bool: showFlavors
strings for which string to display on force-feed success, failure, and eat?
This way you can have your "You wipe your mouth." text.

**Review by whatston3 — APPROVED** [review]
Looks good to me.

Made a small change to the force feed user entity to get it to work (DrinkSystem does it the same now, this one had a strange particularity for some reason).

Happy with this, leaving an approval pending review by others - if there are concerns for the localization strings being sent over the network, seems plausible.

**Pull request merged**